v2/checks/vulnerabilities/check.go

```diff
@@ -19,23 +58,25 @@ func (c *check) SetSBOMClient(sbomClient voucher.SBOMClient) {
 	c.sbomClient = sbomClient
 }
 
-// hasSBOM returns true if the passed image has an SBOM attached
-func (c *check) hasVulnerabilities(i voucher.ImageData) (bool, error) {
+// hasAListedVulnerability returns true if the passed image has an SBOM attached
+func (c *check) hasAListedVulnerability(i voucher.ImageData) ([]string, error) {
+	vulnerabilitiesFound := make([]string, 0, len(c.failOnVulnerabilities))
 	// 1. image reference -> SBOM
-	// 2. SBOM -> look at the "vulnerabilities" key and see if len is greater than zero
-	// Parse the image reference
+	// 2. SBOM -> look at the "vulnerabilities" key and
+	//    see if there are any matching vuln IDs in our list
 	imageName := i.Name()
 	tag := getSBOMTagFromImage(i)
 
 	sbom, err := c.sbomClient.GetSBOM(context.Background(), imageName, tag)
 	if err != nil {
-		// will return false for now
-		// but it is not necessarily correct
-		// TODO: this needs to return something.
-		return true, err
+		return vulnerabilitiesFound, err
```

Comment:
```suggestion
		return nil, err
```

Nit: we haven't written to `vulnerabilitiesFound` yet  - can we make it more explicit that we're returning a zero-value here.

Reply:
nil it is!